src/uwtools/tests/test_cli.py

```diff
@@ -281,22 +281,73 @@ def test__dispatch_forecast_run():
         forecast_model="foo",
     )
     vars(args).update({STR.cfgfile: 1, "forecast_model": "foo"})
-    with patch.object(cli.uwtools.drivers.forecast, "FooForecast", create=True) as m:
+    with patch.object(cli.uwtools.drivers.forecast, "FooForecast", create=True) as module:
         CLASSES = {"foo": getattr(cli.uwtools.drivers.forecast, "FooForecast")}
         with patch.object(cli.uwtools.drivers.forecast, "CLASSES", new=CLASSES):
             cli._dispatch_forecast_run(args)
-    assert m.called_once_with(args)
-    m().run.assert_called_once_with(cycle="2023-01-01T00:00:00")
+    assert module.called_once_with(args)
+    module().run.assert_called_once_with(cycle="2023-01-01T00:00:00")
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        (STR.realize, "_dispatch_rocoto_realize"),
+        (STR.validate, "_dispatch_rocoto_validate"),
+    ],
+)
+def test__dispatch_rocoto(params):
+    submode, funcname = params
+    args = ns()
+    vars(args).update({STR.submode: submode})
+    with patch.object(cli, funcname) as module:
+        cli._dispatch_rocoto(args)
+    assert module.called_once_with(args)
+
+
+def test__dispatch_rocoto_realize():
+    args = ns()
+    vars(args).update({STR.infile: 1, STR.outfile: 2})
+    with patch.object(cli.uwtools.rocoto, "realize_rocoto_xml") as module:
+        cli._dispatch_rocoto_realize(args)
+    assert module.called_once_with(args)
+
+
+def test__dispatch_rocoto_realize_invalid():
+    args = ns()
+    vars(args).update(
+        {
+            STR.infile: 1,
+            STR.outfile: 2,
+        }
+    )
+    with patch.object(cli.uwtools.rocoto, "realize_rocoto_xml", return_value=False):
+        assert cli._dispatch_rocoto_realize(args) is False
+
+
+def test__dispatch_rocoto_validate_xml():
+    args = ns()
+    vars(args).update({STR.infile: 1, STR.infmt: FORMAT.rocoto, STR.verbose: False})
+    with patch.object(cli.uwtools.rocoto, "validate_rocoto_xml") as module:
+        cli._dispatch_rocoto_validate(args)
+    assert module.called_once_with(args)
```

Comment:
This seems to be called like this:

```
uwtools.rocoto.validate_rocoto_xml(input_xml=args.input_file)
```

Reply:
Caught by the cli letting me override too much again, thanks